include/deal.II/lac/lapack_full_matrix.h

```diff
@@ -205,20 +205,31 @@ class LAPACKFullMatrix : public TransposeTable<number>
    *
    * @note Source and destination must not be the same vector.
    *
-   * @note This template only exists for compile-time compatibility with
-   * FullMatrix. Implementation is only available for
-   * <tt>VECTOR=Vector&lt;number&gt;</tt>
+   * @note The template with @tref number2 only exists for compile-time
+   * compatibility with FullMatrix. Only the case @tref number2 = @tref
+   * number is implemented due to limitations in the underlying LAPACK
+   * interface
```

Comment:
So out of curiosity, why do you need this? It will lead to an unhelpful linker error, rather than a compiler error, if you happen to call the function with the wrong template arguments.